### ISSUE TITLE
Update Public Key in Validator Registration Contract to Bytes

### ISSUE THREAD

**Issue by terenc3t** [prysmaticlabs/prysm#562]
The plan is to move on to BLS12-381 signatures where the size of public key on the new curve are 48 bytes.

Currently VRC (validator registration contract) takes type `bytes32` for public key. Using BLS12-381, public key needs to be 48 bytes, I propose we change public key's type to `bytes` and check if it's 48 bytes or revert.

https://github.com/prysmaticlabs/prysm/blob/master/contracts/validator-registration-contract/validator_registration.sol#L5

**Comment by dtco-kimi** [comment]
I'd like to resolve this issue. I'm wondering it's necessary to generate .go and update the test file? 
In `setup()` of the test file, I have no ides how to replace ECDSA key with BLS12-381. Would you please provide me some information?


In my solution, I plan to use 
```
mapping (bytes32 => bool) public usedHashedPubkey;

// function deposit(...)
bytes32 hashedPubkey = keccak256(abi.encodePacked(_pubkey));
require(
        !usedHashedPubkey[hashedPubkey],
        "Public key already used"
);
usedHashedPubkey[hashedPubkey] = true;
```
to replace 
`mapping (bytes32 => bool) public usedPubkey;`
due to `mapping` is not allowed to use dynamic type as a key.

**Comment by KimiWu123** [comment]
I'd like to resolve this issue. I'm wondering it's necessary to generate .go and update the test file?
In `setup()` of the test file, I have no ides how to replace ECDSA key with BLS12-381. Would you please provide me some information?

In my solution, I plan to use

```
mapping (bytes32 => bool) public usedHashedPubkey;

// function deposit(...)
bytes32 hashedPubkey = keccak256(abi.encodePacked(_pubkey));
require(
        !usedHashedPubkey[hashedPubkey],
        "Public key already used"
);
usedHashedPubkey[hashedPubkey] = true;
```
to replace
`mapping (bytes32 => bool) public usedPubkey;`

due to `mapping` is not allowed to use dynamic type as a key.

**Comment by nisdas** [comment]
Hey @KimiWu123 , thanks for looking at this! But I think @IvanTheGreatDev  is already tackling this in #630. EDCSA or BLS sigs are not relevant to this issue, just changing the key size to 48 bytes. Although if I am remembering correctly the issue was unable to be resolved because keys in a mapping unable to be of a dynamic type. 

What you have above looks like a viable solution, @IvanTheGreatDev what do you think about this ?

**Comment by IvanTheGreatDev** [comment]
I was thinking we could split the 48 byte key size into one 32 byte and one 16 byte data field, and then hash the result of that, but yeah we can actually just hash the bytes array they pass in itself :)

**Comment by rauljordan** [comment]
Closed by #630
